src/rebar_fetch.erl

```diff
@@ -61,7 +61,7 @@ download_source(AppDir, Source, State) ->
     catch
         C:T ->
             ?DEBUG("rebar_fetch exception ~p ~p ~p", [C, T, erlang:get_stacktrace()]),
-            throw(?PRV_ERROR({fetch_fail, Source}))
+            ?PRV_ERROR({fetch_fail, Source})
```

Comment:
Ah ok, I revert this change. I think `rebar_fetch:get_resource_type/1` and the error handling of `rebar_fetch` exports needs to be looked at. The specs suggests error tuples can be returned but these aren't handled by callers or an error tuple is never returned but it is handled by callers.